spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskMetrics.java

```diff
@@ -88,41 +92,47 @@
 	 */
 	public static final String TASK_EXCEPTION_TAG = "task.exception";
 
-	private Timer.Sample taskSample;
-
 	private LongTaskTimer.Sample longTaskSample;
 
-	private Throwable exception;
+	private Observation observation;
```

Comment:
In order to make context propagation work, we should create a `Scope`.
Instead of storing the `Observation` here, I think we can open a `Scope` (`observation.openScope()`), store the scope here and close it in `onTaskEnd` (we can also get the observation from the scope: `scope.getCurrentObservation()`).